accounts/ethstore/src/secret_store.rs

```diff
@@ -166,6 +166,7 @@ impl StoreAccountRef {
 	}
 }
 
+#[allow(clippy::derive_hash_xor_eq)] // If they are equal, they have the same address and hence the same hash.
```

Comment:
It is probably better to `derive(Hash)` for `SecretVaultRef` and `SecretVaultRef` and remove the manual hash impl.

Any opinions @svyatonik?

Reply:
I'm not 100% sure (it has been long time ago), but iirc the specific implementation if for the cases when there's a keyfile for the same account in two (actually >1) separate vaults. This has caused issues with UI and signer => probably there's a code somewhere that relies on that property. I can't find the code right now (if it exists), but would advice you to keep the specific `Hash` implementation here.

Reply:
(unless you will be able to follow all the flows of `StoreAccountRef` && verify that it didn't cause any issues)